tests/bluetooth/tester/src/btp_bap.c

```diff
@@ -0,0 +1,206 @@
+/* btp_bap.c - Bluetooth BAP Tester */
+
+/*
+ * Copyright (c) 2021-2022 Nordic Semiconductor ASA
+ * Copyright (c) 2023 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <zephyr/kernel.h>
+
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/audio/audio.h>
+#include <zephyr/bluetooth/audio/pacs.h>
+
+#include "endpoint.h"
+#include "btp/btp.h"
+
+#define CONTROLLER_INDEX 0
+
+#define SUPPORTED_SINK_CONTEXT	(BT_AUDIO_CONTEXT_TYPE_UNSPECIFIED | \
+				 BT_AUDIO_CONTEXT_TYPE_CONVERSATIONAL | \
+				 BT_AUDIO_CONTEXT_TYPE_MEDIA | \
+				 BT_AUDIO_CONTEXT_TYPE_GAME | \
+				 BT_AUDIO_CONTEXT_TYPE_INSTRUCTIONAL)
+
+#define SUPPORTED_SOURCE_CONTEXT (BT_AUDIO_CONTEXT_TYPE_UNSPECIFIED | \
+				  BT_AUDIO_CONTEXT_TYPE_CONVERSATIONAL | \
+				  BT_AUDIO_CONTEXT_TYPE_MEDIA | \
+				  BT_AUDIO_CONTEXT_TYPE_GAME)
+
+#define AVAILABLE_SINK_CONTEXT SUPPORTED_SINK_CONTEXT
+#define AVAILABLE_SOURCE_CONTEXT SUPPORTED_SOURCE_CONTEXT
+
+static struct bt_codec lc3_codec =
+	BT_CODEC_LC3(BT_CODEC_LC3_FREQ_ANY, BT_CODEC_LC3_DURATION_10,
+		     BT_CODEC_LC3_CHAN_COUNT_SUPPORT(1),
+		     40u, 120u, 1u,
+		     (BT_AUDIO_CONTEXT_TYPE_CONVERSATIONAL |
+		     BT_AUDIO_CONTEXT_TYPE_MEDIA));
+
+static struct bt_pacs_cap cap_sink = {
+	.codec = &lc3_codec,
+};
+
+static struct bt_pacs_cap cap_source = {
+	.codec = &lc3_codec,
+};
+
+static int set_location(void)
+{
+	int err;
+
+	err = bt_pacs_set_location(BT_AUDIO_DIR_SINK,
+				   BT_AUDIO_LOCATION_FRONT_CENTER);
+	if (err != 0)
+		return err;
+
+	err = bt_pacs_set_location(BT_AUDIO_DIR_SOURCE,
+				   (BT_AUDIO_LOCATION_FRONT_LEFT |
+				   BT_AUDIO_LOCATION_FRONT_RIGHT));
+	if (err != 0)
+		return err;
+
+	return 0;
+}
+
+static int set_available_contexts(void)
+{
+	int err;
+
+	err = bt_pacs_set_available_contexts(BT_AUDIO_DIR_SOURCE,
+					     AVAILABLE_SOURCE_CONTEXT);
+	if (err != 0)
+		return err;
+
+	err = bt_pacs_set_available_contexts(BT_AUDIO_DIR_SINK,
+					     AVAILABLE_SINK_CONTEXT);
+	if (err != 0)
+		return err;
+
+	return 0;
+}
+
+static int set_supported_contexts(void)
+{
+	int err;
+
+	err = bt_pacs_set_supported_contexts(BT_AUDIO_DIR_SOURCE,
+					     SUPPORTED_SOURCE_CONTEXT);
+	if (err != 0)
+		return err;
+
+	err = bt_pacs_set_supported_contexts(BT_AUDIO_DIR_SINK,
+					     SUPPORTED_SINK_CONTEXT);
+	if (err != 0)
+		return err;
+
+	return 0;
+}
+
+static void pacs_supported_commands(uint8_t *data, uint16_t len)
+{
+	uint8_t cmds[2];
+	struct pacs_read_supported_commands_rp *rp = (void *) cmds;
+
+	(void)memset(cmds, 0, sizeof(cmds));
+
+	tester_set_bit(cmds, PACS_READ_SUPPORTED_COMMANDS);
+
+	tester_send(BTP_SERVICE_ID_PACS, PACS_READ_SUPPORTED_COMMANDS,
+		    CONTROLLER_INDEX, (uint8_t *) rp, sizeof(cmds));
```

Comment:
```suggestion
	struct pacs_read_supported_commands_rp *rp = (void *)cmds;

	(void)memset(cmds, 0, sizeof(cmds));

	tester_set_bit(cmds, PACS_READ_SUPPORTED_COMMANDS);

	tester_send(BTP_SERVICE_ID_PACS, PACS_READ_SUPPORTED_COMMANDS,
		    CONTROLLER_INDEX, (uint8_t *)rp, sizeof(cmds));
```

We generally don't have space between the typecast and the value